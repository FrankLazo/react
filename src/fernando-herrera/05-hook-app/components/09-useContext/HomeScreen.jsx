import { useContext } from 'react';
import UserContext from './UserContext';

const HomeScreen = () =>
{
	const { user } = useContext(UserContext);

	return (
		<div className="min-height-full content has-background-dark p-6 has-text-light">
			<h1 className="has-text-light">Home</h1>
			<hr className="full-rule has-background-grey-dark" />

			<pre>
			{
				JSON.stringify(user, null, 2)
			}
			</pre>
		</div>
	);
};

export default HomeScreen;
