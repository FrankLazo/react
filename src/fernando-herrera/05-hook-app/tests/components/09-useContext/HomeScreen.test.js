import { mount } from 'enzyme';
import HomeScreen from '../../../components/09-useContext/HomeScreen';
import UserContext from '../../../components/09-useContext/UserContext';

describe('<HomeScreen /> tests', () => {
	const user = {
		name: 'Frank',
		mail: 'example@mail.com',
	};

	const wrapper = mount(
		<UserContext.Provider value={{ user }}>
			<HomeScreen />
		</UserContext.Provider>
	);

	test('should show the right component', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
