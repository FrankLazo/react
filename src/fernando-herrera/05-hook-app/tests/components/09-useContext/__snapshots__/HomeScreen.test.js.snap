// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> tests should show the right component 1`] = `
<div
  className="min-height-full content has-background-dark p-6 has-text-light"
>
  <h1
    className="has-text-light"
  >
    Home
  </h1>
  <hr
    className="full-rule has-background-grey-dark"
  />
  <pre>
    {
  "name": "Frank",
  "mail": "example@mail.com"
}
  </pre>
</div>
`;
